### PR TITLE
Implement replaces logic

### LINKED ISSUE #9: Replaces Achievements not defined yet
- not even implemented yet
- replaces one or all?
- what does it even?

### PR DESCRIPTION
**Pull request by frederike-ramin**
Closes #9.
Two TODOs: 
1) Replace only one of the given achievements? Currently I am removing all of them. (Setting amount to 0)
2) Actually delete the row? (couldn't find correct method , so I'll leave it to someone who knows how to google that^^

### CODE REVIEW

**Comment by cmfcmf** [comment]
> Actually delete the row? (couldn't find correct method , so I'll leave it to someone who knows how to google that^^

The method is called `remove(id)` (found here: https://docs.feathersjs.com/api/services).

File: src/hooks/achievement-rule-checker.js [hunk 1]
```diff
@@ -24,9 +24,18 @@ module.exports = function (options = {}) {
...
         }
+
+        for (const replaceName of achievementRule.replaces) {
+          const replacedAchievement = await achievementService.find({
+            query: {
+              user_id: context.data.user_id,
+              name: replaceName
+            }
+          });
+          context.app.service('achievements').remove(replacedAchievement[0]._id);
```
Comment (cmfcmf) [inline src/hooks/achievement-rule-checker.js]: You should probably also `await` this, even if we do nothing with the result. This is to make sure it really is removed before continuing execution.

File: src/hooks/achievement-rule-checker.js [hunk 2]
```diff
@@ -24,9 +24,18 @@ module.exports = function (options = {}) {
...
           achievementService.create({user_id: context.data.user_id, name: achievementRule.name, amount: 1});
         }
+
+        for (const replaceName of achievementRule.replaces) {
+          const replacedAchievement = await achievementService.find({
+            query: {
+              user_id: context.data.user_id,
+              name: replaceName
+            }
+          });
```
Comment (cmfcmf) [inline src/hooks/achievement-rule-checker.js]: `replacedAchievement` can be an empty array if the to-be-replaced achievement was not granted to the user before. The next line will break then.

Comment (frederike-ramin) [reply src/hooks/achievement-rule-checker.js]: But in that case it wouldn't enter the for loop in the first place? `achievementRule.replaces` would be empty and therefore there wouldn't be a `replaceName` to begin with.

**Comment by corinnaj** [comment]
Okay, so that didn't quite work our how I wanted :sweat_smile: But at Travis works now...

**Pull request merged**